File: top-interview-questions-easy/dynamic programming/maxinum-subarray/53.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currentSum, totalSum = nums[0], nums[0]
        for i in range(1, len(nums)):
            currentSum = max(nums[i], nums[i] + currentSum)
            totalSum = max(totalSum, currentSum)
        return totalSum

File: top-interview-questions-easy/dynamic programming/maxinum-subarray/53.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  let currentSum = nums[0]
  let totalSum = nums[0]
  for(var i = 1; i < nums.length; i++) {
    currentSum = Math.max(nums[i], nums[i] + currentSum)
    totalSum = Math.max(totalSum, currentSum)
  }
  return totalSum
}
